source/common/config/well_known_names.cc

```diff
@@ -0,0 +1,109 @@
+  name_regex_pairs.push_back({DYNAMO_PARTITION_ID, "^http(?=\\.).*?\\.dynamodb\\.table(?=\\.).*?\\."
+                                                   "capacity(?=\\.).*?(\\.__partition_id=(\\w{7}))"
+                                                   "$"});
+
+  // http.[<stat_prefix>.]dynamodb.operation.(<operation_name>.)<base_stat> or
+  // http.[<stat_prefix>.]dynamodb.table.[<table_name>.]capacity.(<operation_name>.)[<partition_id>]
+  name_regex_pairs.push_back({DYNAMO_OPERATION, "^http(?=\\.).*?\\.dynamodb.(?:operation|table(?="
+                                                "\\.).*?\\.capacity)(\\.(.*?))(?:\\.|$)"});
+
+  // mongo.[<stat_prefix>.]collection.[<collection>.]callsite.(<callsite>.)query.<base_stat>
+  name_regex_pairs.push_back(
+      {MONGO_CALLSITE,
+       "^mongo(?=\\.).*?\\.collection(?=\\.).*?\\.callsite\\.((.*?)\\.).*?query.\\w+?$"});
+
+  // http.[<stat_prefix>.]dynamodb.table.(<table_name>.) or
+  // http.[<stat_prefix>.]dynamodb.error.(<table_name>.)*
+  name_regex_pairs.push_back(
+      {DYNAMO_TABLE, "^http(?=\\.).*?\\.dynamodb.(?:table|error)\\.((.*?)\\.)"});
+
+  // mongo.[<stat_prefix>.]collection.(<collection>.)query.<base_stat>
+  name_regex_pairs.push_back(
+      {MONGO_COLLECTION, "^mongo(?=\\.).*?\\.collection\\.((.*?)\\.).*?query.\\w+?$"});
+
+  // mongo.[<stat_prefix>.]cmd.(<cmd>.)<base_stat>
+  name_regex_pairs.push_back({MONGO_CMD, "^mongo(?=\\.).*?\\.cmd\\.((.*?)\\.)\\w+?$"});
+
+  // cluster.[<route_target_cluster>.]grpc.[<grpc_service>.](<grpc_method>.)<base_stat>
+  name_regex_pairs.push_back(
+      {GRPC_BRIDGE_METHOD, "^cluster(?=\\.).*?\\.grpc(?=\\.).*\\.((.*?)\\.)\\w+?$"});
+
+  // http.[<stat_prefix>.]user_agent.(<user_agent>.)<base_stat>
+  name_regex_pairs.push_back({HTTP_USER_AGENT, "^http(?=\\.).*?\\.user_agent\\.((.*?)\\.)\\w+?$"});
+
+  // vhost.[<virtual host name>.]vcluster.(<virtual_cluster_name>.)<base_stat>
+  name_regex_pairs.push_back({VIRTUAL_CLUSTER, "^vhost(?=\\.).*?\\.vcluster\\.((.*?)\\.)\\w+?$"});
+
+  // http.[<stat_prefix>.]fault.(<downstream_cluster>.)<base_stat>
+  name_regex_pairs.push_back(
+      {FAULT_DOWNSTREAM_CLUSTER, "^http(?=\\.).*?\\.fault\\.((.*?)\\.)\\w+?$"});
+
+  // listener.[<port>.]ssl.cipher.(<cipher>)
+  name_regex_pairs.push_back({SSL_CIPHER, "^listener(?=\\.).*?\\.ssl\\.cipher(\\.(.*?))$"});
+
+  // cluster.[<route_target_cluster>.]grpc.(<grpc_service>.)*
+  name_regex_pairs.push_back({GRPC_BRIDGE_SERVICE, "^cluster(?=\\.).*?\\.grpc\\.((.*?)\\.)"});
+
+  // tcp.(<stat_prefix>.)<base_stat>
+  name_regex_pairs.push_back({TCP_PREFIX, "^tcp\\.((.*?)\\.)\\w+?$"});
+
+  // auth.clientssl.(<stat_prefix>.)<base_stat>
+  name_regex_pairs.push_back({CLIENTSSL_PREFIX, "^auth\\.clientssl\\.((.*?)\\.)\\w+?$"});
+
+  // ratelimit.(<stat_prefix>.)<base_stat>
+  name_regex_pairs.push_back({RATELIMIT_PREFIX, "^ratelimit\\.((.*?)\\.)\\w+?$"});
+
+  // cluster.(<cluster_name>.)*
+  name_regex_pairs.push_back({CLUSTER_NAME, "^cluster\\.((.*?)\\.)"});
```

Comment:
QQ: will we pick up a some reserved words in stats names as cluster names here?

Reply:
I'm not sure if I'm understanding your question correctly, but I don't think so.  This regex should only be run once, so when it is run, the only string that can be following "cluster." is the actual name of the cluster.  As long as no other regex removes the cluster name before this one, it should work as intended.